tests/trainer/test_trainer.py

```diff
@@ -460,6 +510,54 @@ def on_load_checkpoint(self, _):
         assert next_model.num_on_load_checkpoint_called == 1
 
 
+def test_trainer_properties_resume_from_checkpoint(tmpdir):
```

Comment:
is it possible for us to test this without relying on a callback?

Reply:
yes, I can put all of these inside the lightning module's hooks, but just wanted to keep them separate just like it's done in other tests inside `models/test_restore`.

Reply:
Why not move it to `test_restore`?

Reply:
Because that one is under `models/` and this one is more trainer specific.